addons/account/models/account_move.py

```diff
@@ -1528,12 +1528,29 @@ def _compute_quick_encoding_vals(self):
         for move in self:
             move.quick_encoding_vals = move._get_quick_edit_suggestions()
 
-    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date', 'amount_total')
     def _compute_duplicated_ref_ids(self):
-        move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
+        if self.is_invoice():
+            move_to_duplicate_move = self._fetch_duplicate_invoice_reference()
+        else:
+            move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
         for move in self:
             move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
 
+    def _fetch_duplicate_invoice_reference(self):
+        if self.is_invoice() and self.partner_id and self.invoice_date and self.amount_total:
+            duplicated_ref_ids = self.env['account.move'].search(
+                [
+                    ('company_id', '=', self.company_id.id),
```

Comment:
Question: 
Is the company really necessary ? 
Perf ? better to use a search or a query ?